src/app/auth/index.ts

```diff
@@ -193,13 +195,7 @@ const authResult = NextAuth(() => ({
       return token;
     },
     session({ session, token }) {
-      // delete session.user, that way next-auth will add the value of token to the session
-      // returned from the auth(). This allows us to read accessToken on the server but
-      // not return it from `GET /api/auth/session`.
-      // It isn't ideal though because we can still accidentally pass the session to the client
-      // component. next-auth use react taint api, but it's still experimental so they don't use it
-      delete (session as any).user;
-
+      session.access_token = token.access_token;
```

Comment:
This looks like unnecessary duplication of the access token in the session cookie.

Reply:
The token may update; this is how it should be done based on the documentation.

Reply:
The refresh token and the expiration will update as well, why is it not done the same for them?

Reply:
Because we don't need to access those parameters in the app. 
I am adding `access_token` to the `session` because I use it in the API middleware.

Reply:
While refresh token and expiration of the access token are not used in the API middleware, their validity after update is just as important as with the access token.

The question is why access token has a special treatment including duplication and the others have not given they are all valid.

Reply:
The next auth seems to track the JWT "token" separately regardless 👍 resolved